### ISSUE TITLE
[BUG] CRITICAL risk level unreachable from live model output — RandomForest output ceiling is ~0.873

### ISSUE THREAD

**Issue by 404RiyaFound** [YASHK-arch/NavNER-AI#17]
### Description

Even after the composite-blend fix in #16, `CRITICAL` remains unreachable in practice: the RandomForest regressor's output range is compressed, so the highest landslide score it can emit **anywhere in its input domain** is ~**0.873**, against a `CRITICAL` threshold of **0.85** (Stage 2 PRD §4C).

The practical consequence is that realistic severe NER monsoon conditions classify as `HIGH`, not `CRITICAL`.

Measured via `RiskAssessmentEngine.evaluate_cell`:

| conditions | landslide | flood | level |
|---|---|---|---|
| 42° slope, 260 mm/24h, 95% soil saturation, susceptibility 0.85 | 0.8441 | 0.6766 | **HIGH** |
| domain maximum: 60° slope, 300 mm/24h, 100% saturation, susceptibility 1.0 | 0.8732 | 0.7938 | CRITICAL |

So `CRITICAL` is only reachable in roughly the top 2.7% sliver of the model's achievable output range — effectively only at the extreme corner of the input space, which does not correspond to "a severe but real monsoon event".

**Cause.** `_landslide_model` / `_flood_model` are `RandomForestRegressor`s (`backend/app/risk_engine.py`). A random forest regressor averages predictions across leaves, which shrinks its output toward the training mean — a well-known property. The synthetic training labels are generated from a weighted linear expression and clipped, so the label distribution already concentrates away from 1.0, and tree averaging compresses it further. The result is a model that cannot express the top of its own scale.

**Why it matters.** `CRITICAL` is not a cosmetic label; several behaviours key off it exclusively:

- `ACTION_MAP["CRITICAL"] = "IMMEDIATE_REROUTE"` (`risk_engine.py`) — the mandatory-reroute action per PRD §4C is unreachable from live model output.
- `scheduler.run_risk_evaluation` dispatches the `CRITICAL`-tier alert (`HIGH_PROBABILITY_LANDSLIDE` / `IMMEDIATE_REROUTE`) only when `composite_risk_level == "CRITICAL"`; otherwise it emits `SPEED_RESTRICTION` at the INFORMATIONAL tier, which is *batched hourly* rather than dispatched immediately.
- The Stage 2 PRD's `action_required: "IMMEDIATE_REROUTE"` in the hazard-map contract, and the automated roadblock trigger described for CRITICAL, never fire.

Net effect: the platform can currently never escalate a predicted hazard to its highest severity from real data. Seeded `CRITICAL` cells do display correctly (their scores are hand-authored above 0.85), so the dashboard looks right on a fresh start — the gap only appears after a live evaluation, which makes it easy to miss.

### Steps to Reproduce

1. From `backend/`, probe the model at the maximum of its input domain:

   ```python
   from app.risk_engine import risk_engine
   print(risk_engine.evaluate_cell(
       weather_data={"rainfall_1h_mm": 80.0, "rainfall_24h_mm": 300.0,
                     "soil_saturation_pct": 100.0, "temperature_c": 22.0,
                     "surface_runoff_rate": 100.0},
       terrain_data={"avg_slope_degrees": 60.0, "elevation_meters": 3500.0,
                     "landslide_susceptibility_base": 1.0}))
   ```
   → `landslide_risk_score: 0.8732` — the ceiling. No input produces more.

2. Probe realistic severe conditions (comparable to a Gangtok/Aizawl corridor event):

   ```python
   print(risk_engine.evaluate_cell(
       weather_data={"rainfall_1h_mm": 45.0, "rainfall_24h_mm": 260.0,
                     "soil_saturation_pct": 95.0, "temperature_c": 22.0,
                     "surface_runoff_rate": 40.0},
       terrain_data={"avg_slope_degrees": 42.0, "elevation_meters": 1650.0,
                     "landslide_susceptibility_base": 0.85}))
   ```
   → `composite_risk_level: "HIGH"`.

3. Confirm end to end against live weather:
   ```bash
   curl -X POST http://localhost:8000/api/v1/analytics/evaluate-grid \
     -H 'Content-Type: application/json' -d '{}'
   ```
   → `critical_count: 0`. With #16 applied the run yields `high_count: 10, moderate_count: 10`; before #16 it was `0 high, 20 moderate`.

### Expected Behavior

A severe-but-plausible NER monsoon event on steep terrain should classify `CRITICAL`, so that `IMMEDIATE_REROUTE`, the CRITICAL alert tier, and the automated roadblock trigger are reachable from real data.

Options, roughly in increasing order of correctness:

1. **Calibrate the model's output to the score scale.** Fit an isotonic or Platt calibration mapping raw predictions onto `[0, 1]` so the top of the range is attainable. Contained change, keeps the PRD thresholds intact.
2. **Stop using a random forest to evaluate a deterministic formula.** The labels come from a weighted linear expression, so the forest can only approximate it while adding averaging bias. Evaluating the expression directly removes the compression entirely and is faster — see the related concern that the current model is a formula wrapped in an estimator.
3. **Re-derive thresholds from the observed score distribution** rather than the fixed 0.35/0.65/0.85 cut points. Cheapest, but it deviates from Stage 2 PRD §4C and should be a deliberate, documented decision rather than a silent retune.

Whichever is chosen, add a test asserting that representative severe inputs produce `CRITICAL`. `backend/tests/test_risk_classification.py` (added in #16) already contains `test_severe_monsoon_conditions_reach_at_least_high` and `test_model_output_ceiling_is_documented`, which pin the current ceiling — the latter is written to **fail** once the range widens, so it will flag the fix rather than let it pass silently. Both should be tightened as part of this work.

### Environment

- macOS (Darwin 25.5.0), Python 3.13, scikit-learn 1.9.0, PostGIS 3.4
- `backend/app/risk_engine.py`
- Measured 2026-09-01 against the 20 seeded H3 cells with live Open-Meteo data

### Additional Context

Discovered while writing the regression tests for #16 — the test asserting that severe conditions reach `CRITICAL` failed, which is what surfaced the ceiling. #16 fixes a genuine and separate defect (the composite blend diluted the dominant hazard, so *every* cell came back `MODERATE` and even `HIGH` was unreachable); this issue is the remaining half of the same symptom and is deliberately out of that PR's scope.

Related: the model is trained on synthetic labels generated by a hardcoded formula in the same module, with no historical ground truth and no accuracy metrics. Option 2 above and that broader concern point at the same underlying problem.
